packages/metadata/src/Metadata/v12/toLatest.ts

```diff
@@ -81,12 +89,39 @@ function convertStorage (registry: Registry, { items, prefix }: StorageMetadataV
   });
 }
 
+// generate & register the OriginCaller type
+function registerOriginCaller (registry: Registry, modules: ModuleMetadataV12[]): void {
```

Comment:
can you somehow allow us to insert entries into `KNOWN_ORIGINS`? an additional parameter to registry should be enough.